frontend/js/main.js

```diff
@@ -0,0 +1,2203 @@
+  interleaveCardHeader.role = 'tab';
+
+  interleaveCardHeading = document.createElement("h5")
+  interleaveCardHeading.className = 'mb-0';
+
+  interleaveCardLink = document.createElement("a");
+  interleaveCardLink.innerHTML = `Interleave Table`;
+  interleaveCardLink.className = 'summaryFont';
+  interleaveCardLink.setAttribute("data-toggle", "collapse");
+  interleaveCardLink.setAttribute("href", "#viewInterleave" + index);
+
+  interleaveCardHeading.appendChild(interleaveCardLink);
+  interleaveCardHeader.appendChild(interleaveCardHeading);
+  interleaveCard.appendChild(interleaveCardHeader);
+
+  interleaveCollapse = document.createElement("div")
+  interleaveCollapse.setAttribute("id", 'viewInterleave' + index);
+  interleaveCollapse.className = "collapse summaryCollapse";
+
+  interleaveCollapseCard = document.createElement("div");
+  interleaveCollapseCard.className = "mdc-card mdc-card-content summaryBorder";
+  interleaveCollapseCard.setAttribute('border', '0px');
+
+  if (interleaveApiCallResp.Parent == '') {
+    interleaveApiCallResp.Parent = 'None';
+  }
+
+  interleaveContent = document.createElement('div');
+  interleaveContent.className = 'mdc-card summary-content';
+  interleaveContent.innerHTML = `<span style='font-weight: bold;'>Comment</span>: ${interleaveApiCallResp.Comment} </br> <span style='font-weight: bold;'>Parent</span>: ${interleaveApiCallResp.Parent} </br> <span style='font-weight: bold;'>Possible</span>: ${interleaveApiCallResp.Possible}`;
+  interleaveCollapseCard.appendChild(interleaveContent);
+
+  interleaveCollapse.appendChild(interleaveCollapseCard);
+  interleaveCard.appendChild(interleaveCollapse);
+
+  tableAccContent.appendChild(interleaveCard);
+}
+
+/**
+ * Function to handle spanner table editing
+ *
+ * @param {event} event event generated by clicking edit spanner button
+ * @return {null}
+ */
+const editSpannerHandler = (event) => {
+  if (event.html() == 'Edit Spanner Schema') {
+    $(event[0]).removeAttr('data-toggle');
+  }
+
+  tableNumber = parseInt(event.attr('id').match(/\d+/), 10);
+  uncheckCount[tableNumber] = 0;
+  tableId = '#src-sp-table' + tableNumber + ' tr';
+  event.html("Save Changes");
+  tableColumnNumber = 0;
+  tableCheckboxGroup = '.chckClass_' + tableNumber;
+
+  $(tableId).each(function (index) {
+    if (index == 1) {
+      var temp = $(this).find('.src-tab-cell');
+      temp.prepend(`<span class="bmd-form-group is-filled">
+                      <div class="checkbox">
+                        <label>
+                          <input type="checkbox" id='chckAll_${tableNumber}' value="">
+                          <span class="checkbox-decorator"><span class="check" style='margin-left: -7px;'></span><div class="ripple-container"></div></span>
+                        </label>
+                      </div>
+                    </span>`)
+    }
+    $('#chckAll_' + tableNumber).prop('checked', true);
+    $('#chckAll_' + tableNumber).click(function () {
```

Comment:
This  selector can be pulled out into variable as it's repeating.

`const checkAllTableNumber = $('#chckAll_' + tableNumber);`

Reply:
Agreed.. code changes done accordingly